File: mobile/lib/models/premium_bean.dart
import 'package:json_annotation/json_annotation.dart';

part 'premium_bean.g.dart';

@JsonSerializable()
class PremiumBean {
  @JsonKey(name: '_id')
  final String? id;
  final String beanId;
  final String? kicker;
  final String titleMain;
  final String titleSub;
  final String desc;
  final List<String> pills;
  final String image;
  final double? imgScale;
  final double? imgX;
  final int sortOrder;
  final bool active;
  final String? productId;

  PremiumBean({
    this.id,
    required this.beanId,
    this.kicker,
    required this.titleMain,
    required this.titleSub,
    required this.desc,
    required this.pills,
    required this.image,
    this.imgScale,
    this.imgX,
    this.sortOrder = 0,
    this.active = true,
    this.productId,
  });

  // Convenience getters to match old usage
  String get name => '$titleMain $titleSub';
  String get origin => titleMain;
  String get description => desc;

  factory PremiumBean.fromJson(Map<String, dynamic> json) =>
      _$PremiumBeanFromJson(json);
  Map<String, dynamic> toJson() => _$PremiumBeanToJson(this);
}

File: backend/models/PremiumBean.js
const mongoose = require('mongoose');

const PremiumBeanSchema = new mongoose.Schema({
  beanId: { type: String, required: true, unique: true }, // e.g., "jamaica", "australia"
  kicker: { type: String, default: 'Coffee Beans World • Premium Collection' },
  titleMain: { type: String, required: true },
  titleSub: { type: String, required: true },
  desc: { type: String, required: true },
  pills: [{ type: String }],
  image: { type: String, required: true },
  cloudinaryPublicId: { type: String },
  imgScale: { type: Number, default: 1.0 },
  imgX: { type: Number, default: 0 },
  sortOrder: { type: Number, default: 0 },
  active: { type: Boolean, default: true },
  // Link to a product for direct navigation
  productId: { type: mongoose.Schema.Types.ObjectId, ref: 'Product' },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('PremiumBean', PremiumBeanSchema);
